### PR TITLE
Fix/support for ReduceLROnPlateau scheduler

### PR DESCRIPTION
**Pull request by pennfranc**
Addresses #707. This only adds limited support for `ReduceLROnPlateau` in the sense that it only takes the training loss as `metrics` argument to its `step()` method. To add, say, validation loss we would have to change a few more things since the lr_scheduler is called at every epoch of the training loop, whereas new validation losses are not necessarily computed everytime. I guess it's not worth spending too much time on this considering the new PTL refactoring.

### CODE REVIEW

**Review by hrzn — APPROVED** [review]
Thanks!
